parser-typechecker/unison/Main.hs

```diff
@@ -164,16 +201,32 @@ runTranscripts inFork mcodepath args = do
                                          (FP.takeExtension md)
               writeUtf8 out mdOut
               putStrLn $ "💾  Wrote " <> out
-        wat -> putStrLn $ "Unrecognized command, skipping: " <> wat
-      PT.putPrettyLn $
+        wat -> 
+              PT.putPrettyLn $ P.callout "❓" (
+                P.lines [
+                  P.indentN 2 "Unrecognized command, skipping:", "",
+                  P.indentN 2 $ P.string wat])
+      pure True
+    [] -> 
+      pure False
+
+runTranscripts :: Bool -> Bool -> Maybe FilePath -> [String] -> IO ()
+runTranscripts inFork keepTemp mcodepath args = do
+  transcriptDir <- prepareTranscriptDir inFork mcodepath
+  completed <- runTranscripts' mcodepath transcriptDir args
+  when completed $ do
+    unless keepTemp $ cleanup transcriptDir
+    when keepTemp $ PT.putPrettyLn $
         P.callout "🌸" (
           P.lines [
             "I've finished running the transcript(s) in this codebase:", "",
             P.indentN 2 (P.string transcriptDir), "",
             P.wrap $ "You can run"
                   <> P.backticked ("ucm -codebase " <> P.string transcriptDir)
                   <> "to do more work with it."])
-    [] -> do
+
+  unless completed $ do 
+      unless keepTemp $ cleanup transcriptDir
```

Comment:
I broke off two smaller functions off of the original `runTranscripts` with no changes in functionality. Deletion of the temporary directory happens right before the end.